rules/framework/vfs_overlay.bzl

```diff
@@ -57,7 +58,18 @@ def _build_subtrees(paths, vfs_prefix):
     for path_info in paths:
         path = path_info.framework_path
 
-        parts = path.split("/")
+        # Avoid calling `.split` when possible for better performance
+        parts = []
+        parts_len = 0
+        if "/" in path:
+            parts = path.split("/")
```

Comment:
Making this faster is good but a larger problem is that we have to call this a lot of times right now - which is why this shows up in the perf. It looks like this PR makes a decent savings however - it still has a lot of room to improve.

Can we either:
1. rework `build_subtress` / related data structures to memoize more or all o fthis? 
or:
2. if 1. isn't possible - then cache the `split` call here to do it one time ever https://github.com/bazel-ios/rules_ios/pull/772/files#diff-450b1330485549712d2fd4e65b2b3fd0a7692fe8f4ec43a0b7b60bbb739313afR141

Pt1 seems ideal - and fixing this will help improve _bazel_ bootstrapping time after a checkout a bit - but after it's computed once it's less of an issue

Reply:
I don't want to bloat this PR with more scope atm but I do think your suggestions make sense. I'm doing some small optimizations as I work on something else so not planning to spend much more time on this one for now.

That said, there's def more room for improvement here and will keep you suggestions in mind next time I tackle this.